content/docs/user-guide/assets/pipeline/asset-dependencies-and-identifiers.md

```diff
@@ -26,12 +32,12 @@ Product dependencies contain data on the relationships between product assets. P
 
 ## Asset identifiers
 
-Source assets are identified by a Universally Unique Identifier (UUID). The source asset UUID is generated based on the file name and scan directory relative path. This ensures the source asset UUID is identical on any host platform machine for the project.
-
+Source assets are identified by a Universally Unique Identifier (UUID). This is also referenced as a Globally Unique Identifier (GUID) in some parts of O3DE. The source asset UUID is generated based on the file name and scan directory relative path. This ensures the source asset UUID is identical on any host platform machine for the project.
 
-Product assets are identified by a combination of the UUID for the source asset and a sub ID based on the fingerprint of the **Asset Builder** that produced the product asset. This ensures each product asset ID is deterministic and unique, and that references to these product assets can be maintained when assets are updated. In the rare event of a product asset ID collision, Asset Processor notifies you that an asset already exists with the product ID so you can resolve the issue.
+Product assets are identified by a combination of the UUID for the source asset and a sub ID defined by the **Asset Builder** that produced the product asset. It is the responsibility of the builder author to make sure their sub IDs are deterministic and unique, so that references to these product assets can be maintained when assets are updated. When authoring a builder, keep in mind that other builders may process the same source asset and also generate products, which is one area sub ID collisions can occur. When authoring a builder, try to keep sub IDs as consistent as possible across changes to a source asset, otherwise downstream references, such as a prefab file referencing a product asset via asset ID, will break if that sub ID no longer resolves, or resolves to a different product asset. In the rare event of a product asset ID collision, Asset Processor notifies you that an asset already exists with the product ID so you can resolve the issue.
+
+Note that source assets are not specific to platforms, but product assets are unique across platforms. If you are authoring a builder that outputs product assets differently for different platforms, you should consider how you will access these product assets at runtime. Most of the time the product asset differences are performance optimizations per platform, and in that case it's recommended that the same sub ID is used for similar assets for each platform. For example, the source asset _dev_purple.tif generates a streamingimage file _dev_purple.tif.streamingimage for all platforms, which has the sub ID 1000 on all platforms. This means that any other content referencing this file can reference that sub ID and get the same functional image regardless of which platform the project is running on.
```

Comment:
Maybe this should be in a note callout? Some format/style fixes as well.

{{< note >}}
Product assets are unique across target platforms. The differences for each target platform most often are platform performance optimizations.

If you author a builder that outputs product assets for different target platforms, you should consider how the product assets are accessed at runtime.  It's recommended to use the same sub ID for similar assets for each target platform. For example, the source asset `_dev_purple.tif` generates a streamingimage file `_dev_purple.tif.streamingimage` , which has the sub ID 1000, for all platforms. This means that any other content that references `_dev_purple.tif.streamingimage` can reference the sub ID and get the same functional image on each target platform.
{{< /note >}}